libcextract/SymbolExternalizer.cpp

```diff
@@ -71,142 +87,161 @@ using namespace llvm;
 /* IntervalTree.  */
 using namespace Intervals;
 
-// For debugging purposes.
-#if 0
-extern "C" void Debug_Range(const SourceRange &range)
-{
-  llvm::outs() << PrettyPrint::Get_Source_Text_Raw(range) << '\n';
-}
-
-#define DEBUG_RANGE(x) do { \
-  llvm::outs() << "line " << __LINE__ << "  "; \
-  Debug_Range(x); \
-  } while (0)
-#endif
-
-/** Define a Visitor just to update TypeOfType instances:
- *
- *  Kernel code sometime does things like this:
- *
- *   typeof(symbol_to_externalize) x;
- *
- * this bizarre constructs come from macros, which sometimes explodes because
- * clang-extract is unable to determine which part of it generates the Decl
- * in question.
- *
- * Now we have to update those typeofs, but there is no easy way of parsing Types
- * to get to the Expr.  But the RecursiveASTVisitor knows how to do it, so we
- * hack the class in order to setup a call to
- * FunctionUpdate::Update_References_To_Symbol and update those references.
- *
- */
 class TypeUpdaterVisitor : public RecursiveASTVisitor<TypeUpdaterVisitor>
```

Comment:
Please rename this visitor to something more meaningful. This doesn't only update Types, as it is now responsible for almost all of the externalization logic.

How about `ExternalizerVisitor` ?